### PR TITLE
(VDB-1017) Retry log extraction on error

### PR DESCRIPTION
**Pull request by rmulhol**
- With constant retry interval and max consecutive errors limit

Looking to get some early feedback on this. I think we could merge as is and resolve the issue that's the impetus for this story, but some ideas I'd consider adding include:
- applying the same retry logic to log delegation, and perhaps the storage watcher
- extracting a separate namespace for handling retries, if we do the above
- refactoring the `compose` and `execute` logic, so that we don't have to modify event watcher initialization in two places
- making the max consecutive errors and retry interval configurable from the cli
- refactoring the event watcher tests, attempting to obviate the need for a sentinel error (`errExecuteClosed`)

Hoping to make sure the fundamental approach seems sensible to folks before tossing in all of that :p

### CODE REVIEW

**Comment by rmulhol** [comment]
Update:
- retry logic now applies to both log extraction and delegation.
- max retries and retry interval now configurable via the CLI.
- still using a sentinel error in the event watcher tests, but they are greatly simplified by the removal of async assertions.
- `compose` and `execute` logic now each defined in one place (though future work could pull it out into a tested namespace).
- not planning to apply retries to the storage watcher in this PR because the storage watcher generally retries on default. This is a bit of a ball of wax, though - we don't retry on a fetch error and there are perhaps places where we ought to more carefully handle errors (e.g. failed queueing of a diff after failed execution). Thinking it ought to be addressed by a separate PR.
- Not extracting retry handling to a separate namespace, since it's not clear the approach used in the event watcher is applicable elsewhere.

So, in a nutshell, I'm feeling about ready to merge.
